app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,12 +1,29 @@
 package com.example.ithaca_transit_android_v2
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import com.example.ithaca_transit_android_v2.networking.NetworkUtils
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+
 
 class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        // TODO (lesley): just for testing purposes - replace with rxjava
+        runBlocking {
```

Comment:
not sure what blocking means, i hope to use the rxjava with states

Reply:
Blocking pretty much halts any progress on whichever thread the code is currently being run on until the current function is done. So in this context, any code that comes after the `runBlocking { ... }` block will not be run until we hit the `Log.d("testing-final", deferred.toString())` line. Seems like there's an async function nested inside this block as well but we're calling await so it'll still block? Not sure what's going on but I don't think we wanna do anything related to network requests blockingly. 

Here's also a reference that talks about different threading methods in Kotlin w coroutines: https://medium.com/@elye.project/differentiating-thread-and-coroutine-launch-runblocking-in-kotlin-28219506c002

Reply:
the coroutine is temporary, i just didn't have time to look into rxjava. rxjava is great with controlling the ui depending on if the networking call is successful or not, and it's very good at cancelling failed calls or preventing calls from running if the app is closed

Reply:
Sounds good. J wanted to make note of it thats all

Reply:
aka i'm hoping to add enough error catching to avoid blocking but i'm hoping to do it with rxjava - this is just to make sure that the networking call is return the right value for testing purposes (and for the other android devs to use if they need to do any testing)